crates/walrus-service/src/node.rs

```diff
@@ -658,17 +658,54 @@ impl StorageNode {
             node_params.num_checkpoints_per_blob
         );
 
+        let contract_config = config
+            .sui
+            .as_ref()
+            .expect("Sui config must be provided")
+            .contract_config
+            .clone();
+        let client_config = crate::client::ClientConfig {
+            contract_config,
+            exchange_objects: vec![],
+            wallet_config: None,
+            communication_config: Default::default(),
+            refresh_config: Default::default(),
+        };
+        let sui_read_client = config
```

Comment:
Suggestion: looks like `latest_certified_event_blob` taken in is an option in `EventBlobWriterFactory::new` and if we don't provide it, we continue with normal event processing. In the topic of not blocking node starting up when no fullnode is available, let's don't return error here, but instead use Non in `latest_certified_event_blob`.